libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2256,6 +2297,18 @@ void SMTEncoder::resetStateVariables()
 	m_context.resetVariables([&](VariableDeclaration const& _variable) { return _variable.isStateVariable(); });
 }
 
+void SMTEncoder::resetMemoryVariables()
+{
+	m_context.resetVariables([&](VariableDeclaration const& _variable) { return _variable.referenceLocation() == VariableDeclaration::Location::Memory; });
+}
+
+void SMTEncoder::resetStorageVariables()
+{
+	m_context.resetVariables([&](VariableDeclaration const& _variable) {
+		return _variable.referenceLocation() == VariableDeclaration::Location::Storage || _variable.isStateVariable();
+	});
```

Comment:
I think this is actually the intended behavior of  `resetStateVariables`. So maybe remove `resetStateVariables` and replace by `resetStorageVariables` wherever it's used?

Reply:
As agreed, this will be a separate PR.